src/mol-io/reader/mol/parser.ts

```diff
@@ -28,6 +40,10 @@ export interface MolFile {
         readonly atomIdxB: Column<number>,
         readonly order: Column<number>
     }
+    readonly formalCharges: {
+        readonly atomIdx: Column<number>;
+        readonly charge: Column<number>;
+    }
 }
 
 export function handleAtoms(tokenizer: Tokenizer, count: number): MolFile['atoms'] {
```

Comment:
In `const x = TokenBuilder.create(tokenizer.data, count * 2);`
@dsehnal Why allocate space that is twice the size of the atom count?

Reply:
Start and end index for each token. E.g. it doesn't do "substr" during the parsing, but just passes down the original string and indices into that string. This allows for faster parsing of numbers down the line for example.